### PR TITLE
Google login

### PR DESCRIPTION
**Pull request by KaviHarjani**
Google login official plugin, for directly signing in with logged-in accounts without entering credentials again

### CODE REVIEW

File: lib/webview.dart [hunk 1]
```diff
@@ -16,7 +17,14 @@ class _ScreenerAppState extends State<ScreenerApp> {
   late WebViewController controller;
   final _razorpay = Razorpay();
   var options = {};
-  late bool isLoading = false;
+  late final String _screenerHomeUrl =
+      widget.debug ? "http://10.0.2.2:8000" : 'https://www.screener.in';
+  late String paymentUrl = "'$_screenerHomeUrl/payment/capture/'";
+  late String googleUrl = "'$_screenerHomeUrl/login/flutter_auth/'";
```
Comment (manujagobind) [inline lib/webview.dart]: can be renamed to googleLoginUrl

File: lib/webview.dart [hunk 2]
```diff
@@ -31,31 +39,74 @@ class _ScreenerAppState extends State<ScreenerApp> {
...
+    if (googleUser.length > 1) {
+      await _googleSignIn.disconnect();
+    }
+  }
+
+  Future<void> _handleSignIn() async {
+    try {
+      var user = await _googleSignIn.signIn();
+      if (user != null) {
+        googleUser = "${user.displayName}";
```
Comment (manujagobind) [inline lib/webview.dart:78]: var missing

File: lib/webview.dart [hunk 3]
```diff
@@ -92,47 +143,54 @@ class _ScreenerAppState extends State<ScreenerApp> {
...
+            body: WebView(
+              initialUrl: _screenerHomeUrl,
+              javascriptMode: JavascriptMode.unrestricted,
+              userAgent: _proxyUserAgent,
+              onWebViewCreated: (controller) {
+                this.controller = controller;
+              },
+              zoomEnabled: false,
+              navigationDelegate: (NavigationRequest request) {
+                if (request.url.endsWith('login/google/')) {
```
Comment (manujagobind) [inline lib/webview.dart:155]: Use double quotes with strings for consistency.

File: android/app/build.gradle [hunk 4]
```diff
@@ -66,7 +72,7 @@ android {
    }
     buildTypes {
         release {
-            signingConfig signingConfigs.release
+            signingConfig signingConfigs.debug
```
Comment (KaviHarjani) [inline android/app/build.gradle:75]: This is just for test, will switch to release in production

File: android/app/build.gradle [hunk 5]
```diff
@@ -49,14 +49,20 @@ android {
...
         versionName flutterVersionName
     }
 
    signingConfigs {
+       debug {
+           keyAlias 'androiddebugkey'
+           keyPassword 'Debug@123'
+           storeFile file('debugkeystore.jks')
+           storePassword 'Debug@123'
+       }
```
Comment (KaviHarjani) [inline android/app/build.gradle:65]: debug keys for google login

**Comment by manujagobind** [comment]
Can we clean this PR (or open a new one) that has just the commits for Google Login?

**Pull request closed without merging**